src/gentropy/dataset/summary_statistics.py

```diff
@@ -59,34 +58,30 @@ def window_based_clumping(
         self: SummaryStatistics,
         distance: int = 500_000,
         gwas_significance: float = 5e-8,
-        baseline_significance: float = 0.05,
-        locus_collect_distance: int | None = None,
+        collect_locus: bool = False,
+        collect_locus_distance: int | None = None,
     ) -> StudyLocus:
-        """Generate study-locus from summary statistics by distance based clumping + collect locus.
+        """Generate study-locus from summary statistics using window-based clumping.
+
+        For more info, see [`WindowBasedClumping`][gentropy.method.window_based_clumping.WindowBasedClumping]
 
         Args:
             distance (int): Distance in base pairs to be used for clumping. Defaults to 500_000.
             gwas_significance (float, optional): GWAS significance threshold. Defaults to 5e-8.
-            baseline_significance (float, optional): Baseline significance threshold for inclusion in the locus. Defaults to 0.05.
-            locus_collect_distance (int | None): The distance to collect locus around semi-indices. If not provided, locus is not collected.
+            collect_locus (bool): Whether to collect locus around semi-indices. Defaults to False.
+            collect_locus_distance (int | None): The distance to collect locus around semi-indices. If not provided, `distance` will be used.
 
         Returns:
-            StudyLocus: Clumped study-locus containing variants based on window.
+            StudyLocus: Clumped study-locus optionally containing variants based on window.
         """
-        return (
-            WindowBasedClumping.clump_with_locus(
-                self,
-                window_length=distance,
-                p_value_significance=gwas_significance,
-                p_value_baseline=baseline_significance,
```

Comment:
I am not sure if it is a good idea to drop `baseline_significance` at this point. OK, we can filter data later, but I assume it is cheaper to do it now than filtering the nested data structure of locus. We can keep the default 1 meaning no drop is done though.

Reply:
I was thinking about this. In the [current function](https://github.com/opentargets/gentropy/blob/1095c67a8d9d713615bc8890a5c1926b53815e48/src/gentropy/method/window_based_clumping.py#L182), `baseline_significance` is used to filter the SummaryStatistics at the very top using `summary_stats.pvalue_filter(p_value_significance)`. The `p_value_significance` is no longer used anywhere down the function.

In a sense, this would be the same as if we filter the input summary statistics using `summary_stats.pvalue_filter(p_value_significance)` and pass them to the `clump` function. Everything should work the same. I have a perception that the 2 things are independent and we will increase modularity by keeping them separate, but I am happy to revert if you see a tight coupling between the 2 things.

Reply:
Good point and in a sense it would even work better if the locus collection is a method for StudyLocus